### ISSUE TITLE
Incorrect variable used in function play

### ISSUE THREAD

**Issue by jerome93** [mancuniancol/plugin.video.t411#4]
Hello !
In function play there is a blocking error in the following code:
settings.debug("PlayMedia(%s)" % path)
xbmc.executebuiltin("PlayMedia(%s)" % path)

You should use variable link instead of path.

Also would it be possible to add the same torrent clients as in classier viewer ? I see that they are already managed in play() but not in the settings of the plugin.

Have a nice day
Jérôme

**Comment by mancuniancol** [comment]
I don't understand your comment. The path is local name of the torrent. 1.torrent.  What kind the error are you having?

**Comment by mancuniancol** [comment]
Because it is a private site. I need to download the torrent to keep alive the connection with the credentials for your login. If I send the url the player will be blocked.

**Comment by jerome93** [comment]
You initiate the variable link like this (example for Torrenter, which is
the one I use, but you did it for all) :

elif settings.value["plugin"] == "Torrenter":
        link = 'plugin://plugin.video.torrenter/?action=playSTRM&url=' +
path + \
               '&not_download_only=True'

And instead of using variable link, which works, you use variable path,
which seems to give an error and doesn't play the file. Here is what I get
in kodi.log:

23:10:24 9523.501953 T:1957589552   ERROR: Playlist Player:
skipping unplayable item: 0, path [plugin://plugin.video.t41
1/play/https%3A%2F%2Fwww.t411.ch%2Ftorrents%2Fdownload%2F%3F
id%3D5031886]
23:10:25 9523.793945 T:1524593696   ERROR: Open - error prob
ing input format, /home/osmc/.kodi/temp/1.torrent
23:10:25 9523.794922 T:1524593696   ERROR: OpenDemuxStream -
Error creating demuxer
23:10:25 9523.795898 T:1957589552   ERROR: PlayMedia could n
ot play media: /home/osmc/.kodi/temp/1.torrent

If you want I can send you the difference between what is in the repo and
what I did to make the plugin work on my setup.

**Comment by jerome93** [comment]
Or maybe I just have bad settings on my system leading the plugin to crash

**Comment by mancuniancol** [comment]
Well, I've just tested in windows and openelec. But the idea is to use the path, because the viewer had downloaded the torrent. That is the trick that I use to make the private site works. If you can use link, that means that t411 changed and it is allowing anonymous downloading.

**Comment by mancuniancol** [comment]
But, I want to confirm that it is not an error. It is possible that your system doesn't allow to save the torrent. Could you check it for me? If you use that local torrent in a client torrent. It supposes to download the video without problem.

**Comment by jerome93** [comment]
Yes the torrent file 1.torrent works correctly. And I think we still need
it.

BTW here is the url used now that I replaced the variable:
23:34:02 394.430420 T:1525675040  NOTICE: [T411 viewer] Play
Media(plugin://plugin.video.torrenter/?action=playSTRM&url=/home/osmc/.kodi/temp/1.torrent&not_download_only=True)

So still from the local file.

**Comment by jerome93** [comment]
I just checked your first commit and you were using the link variable :

+ settings.debug("PlayMedia(%s)" % link)+
xbmc.executebuiltin("PlayMedia(%s)" % link)

In commit 4541723b you changed it and since it, it stopped working on my
OSMC.

**Comment by mancuniancol** [comment]
Ic, well. I will check next week when I have more time. I will leave the issue opened as reminder.
